Stock kernel does not support Quick Charge 2.0 technology.
I have tested phone behaviour with up to 2.5A chargers and unlocked this feature.
---

--- a/include/linux/usb/msm_hsusb.h
+++ b/include/linux/usb/msm_hsusb.h
@@ -104,7 +104,7 @@
 	SNPS_FEMTO_PHY,
 };
 
-#define IDEV_CHG_MAX	1500
+#define IDEV_CHG_MAX	2500
 #define IDEV_CHG_MIN	500
 #define IUNIT		100
 
